System/Secretary/DeskEventManager.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * Manages event handlers for an event aware class
+ */
+import { DeskEvent } from "./DeskEvent";
+
+class DeskEventItem {
+    id: number;
+    deskEvent: DeskEvent;
+    constructor(id: number, deskEvent: DeskEvent) {
+        this.id = id;
+        this.deskEvent = deskEvent;
+    }
+}
+
+class DeskEventInfo {
+    id: number;
+    constructor(id: number) {
+        this.id = id;
+    }
+}
+
+export class DeskEventManager {
+    /**
+     * DeskEvent array
+     */
+    private deskEventItems: DeskEventItem[] = [];
+    private lastIndex: -1;
+
+    add(target: Element | Document | Window, method: string, evtFunc: (this: Element, ev: any) => any): DeskEventInfo {
+        const deskEvent = this.deskEventItems.find(
+            (x) => x.deskEvent.target === target && x.deskEvent.method === method && x.deskEvent.evtFunc === evtFunc
+        );
+        if (deskEvent) {
+            return new DeskEventInfo(deskEvent.id);
+        }
+        this.deskEventItems.push(new DeskEventItem(++this.lastIndex, new DeskEvent(target, method, evtFunc, true)));
+        return new DeskEventInfo(this.lastIndex);
+    }
+
+    stop(id: number) {
```

Comment:
I think we said we wanted to use strings for the ids. I think internally we should treat them as incrementing numbers, but the API we expose should treat them as strings.